backend/capellacollab/sessions/operators/k8s.py

```diff
@@ -668,28 +776,91 @@ def _create_service(
                 labels={"app": name},
                 annotations={
                     "prometheus.io/scrape": "true",
-                    "prometheus.io/path": "/metrics",
-                    "prometheus.io/port": "9118",
+                    "prometheus.io/path": prometheus_path,
+                    "prometheus.io/port": f"{prometheus_port}",
                 },
             ),
             spec=client.V1ServiceSpec(
                 ports=[
                     client.V1ServicePort(
-                        name="rdp", protocol="TCP", port=3389, target_port=3389
-                    ),
-                    client.V1ServicePort(
-                        name="metrics",
+                        name=name,
                         protocol="TCP",
-                        port=9118,
-                        target_port=9118,
-                    ),
+                        port=port,
+                        target_port=port,
+                    )
+                    for name, port in ports.items()
                 ],
                 selector={"app": deployment_name},
                 type="ClusterIP",
             ),
         )
         return self.v1_core.create_namespaced_service(namespace, service)
 
+    def _create_ingress(self, id, path: str, port_number: int):
+        ingress = client.V1Ingress(
+            api_version="networking.k8s.io/v1",
+            kind="Ingress",
+            metadata=client.V1ObjectMeta(
+                name=id,
+                annotations={"ingress.kubernetes.io/ssl-redirect": "false"},
+            ),
+            spec=client.V1IngressSpec(
+                rules=[
+                    client.V1IngressRule(
+                        host=host,
+                        http=client.V1HTTPIngressRuleValue(
+                            paths=[
+                                client.V1HTTPIngressPath(
+                                    path=path,
+                                    path_type="Prefix",
+                                    backend=client.V1IngressBackend(
+                                        service=client.V1IngressServiceBackend(
+                                            name=id,
+                                            port=client.V1ServiceBackendPort(
+                                                number=port_number
+                                            ),
+                                        )
+                                    ),
+                                )
+                            ]
+                        ),
+                    )
+                ]
+            ),
+        )
+        return client.NetworkingV1Api().create_namespaced_ingress(
+            namespace, ingress
+        )
```

Comment:
Please move the `NetworkingV1Api` to the `__init__` (e.g., `self.v1_networking = client.NetworkingApi(api_client=self.client)`), so we do not create a new one each time we create a resource